macOS: Richer app menu to enable keyboard navigation

roop opened the pull request:
Adresses #331.

This PR:
  - Disables default keyboard handling in `NSTableView`s in main screen and search screen
  - Introduces a 'Server' app menu with menu items for navigating the server list and selecting a server. It also has menu items for selecting profiles and toggling the VPN, but those are applicable only when in the connection screen.

The menus in each screen look like this:
  - Main screen:
    <img width="288" alt="when_in_main_screen" src="https://user-images.githubusercontent.com/554/125509289-b0f0525e-fd29-4710-a428-e8b784ee25dc.png">
  - Search screen:
    <img width="287" alt="when_in_search_screen" src="https://user-images.githubusercontent.com/554/125509331-2b506b72-7951-4dda-aa09-7b28f4603cdd.png">
  - Connection screen: 
    <img width="304" alt="when_in_connection_screen" src="https://user-images.githubusercontent.com/554/125509369-988ea1c4-822b-4d04-80ed-6f3648771756.png">


Why:

  - When in the search screen, we want the search field to have the focus all the time, so that the user can type-to-search anytime. Therefore, we can't have the tableview getting focus and receiving key events. So, we can't make use of the search screen tableview's default keyboard handling.
  - The app menu is used for navigating the server list and other operations because it enables the user to discover the keyboard shortcuts by exploring the menu

roop commented:
@fkooman @efef Can you please review the UX of the menus?

fkooman commented:
This below is just a suggestion, we can probably make it better still!

The menu items are not really clear in what they do. Begin connecting... to what? Toggle VPN? What does this do? Maybe we can find a better label for it (once I know what it does)

Perhaps you need "fold out / pull out" menu's that open a sub menu when choosing "Connect".

```
Connect to...
   - Radboud Universiteit
   - vpn.tuxed.net
   - ....
   - 
```

If there is only one option, you can obviously do something like "Connect to Radboud Universiteit" (if that fits) or "Connect to VPN".

fkooman commented:
If you extend the context menus just a tiny bit more, there is no use for the main UI at all anymore ;-)

roop commented:
@fkooman These menu items apply on something in the UI, and would make sense only in the context of what's shown in the UI. The "Server" menu is like the "Card" menu in Contacts.app, or like the "Message" menu in Mail.app -- the commands in the menu apply to the current item in the UI.

So, "Begin Connecting" starts the connection flow for the current server, like below:
<img width="632" alt="Screenshot 2021-07-14 at 1 41 59 AM" src="https://user-images.githubusercontent.com/554/125518707-bcb323ca-34bf-481a-b65c-db448c8c24b6.png">

And "Toggle VPN" toggles the VPN switch, like below:
<img width="633" alt="Screenshot 2021-07-14 at 1 43 23 AM" src="https://user-images.githubusercontent.com/554/125518785-b5f28e4a-ea61-4fdf-8869-b59e3e01568b.png">

It wouldn't be a good idea to have a Connect to ... > vpn.tuxed.net menu in the app menu, but we do have that in the status menu.

fkooman commented:
Ah! And why not "Connect" instead of "Begin connecting"? And "Enable" or "Disable" instead of "Toggle VPN" (depending on the state)?

roop commented:
I didn't want to say "Connect" because whether we'd connect or not depends on how many profiles we get, and whether we get a profile that we've already seen.

I used "Toggle VPN" because Cmd+T is easier to associate with toggling, rather than "Enable" / "Disable". And I didn't want to have both "Enable" and "Disable" items, with only one usable at a time.

johankool commented:
> I didn't want to say "Connect" because whether we'd connect or not depends on how many profiles we get, and whether we get a profile that we've already seen.

The way you indicate that there will be more steps first is to add the 3 dots. So "Connect…" would make sense here.

The command-P for select profile is odd. That one is really reserved for printing. You'll never want to do that in this app, but it is still an unusual key combo to use for anything other than printing. A key combo to open a pulldown is 'odd' anyway. Not sure what the 'right' way is though.

roop commented:
@johankool:

> The way you indicate that there will be more steps first is to add the 3 dots. So "Connect…" would make sense here.

According to the [HIG](https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-anatomy/):

> **Use an ellipsis whenever choosing a menu item requires additional input from the user.** The ellipsis character (…) means a dialog or separate window will open and prompt the user for additional information or to make a choice.

So I'm not entirely sure if we should say "Connect..." here. I used ellipsis for "Select Profiles..." which doesn't exactly fit the HIG description either, so I'm not fully against "Connect...".

> The command-P for select profile is odd. That one is really reserved for printing. You'll never want to do that in this app, but it is still an unusual key combo to use for anything other than printing.

Good catch. Any suggestions for the key combo we can use instead?

> A key combo to open a pulldown is 'odd' anyway. Not sure what the 'right' way is though.

It's indeed a little odd, but I can't think of a better way. We could do Cmd+1 / Cmd+2 / Cmd+3 for selecting first / second / third profiles, but that's probably more weird?

roop commented:
@johankool:

- Changed "Begin Connecting" to "Connect"
- Changed keyboard shortcut of "Select Profile..." from Cmd+P to Cmd+L
- Rebased to master
